js/core.js

```diff
@@ -166,20 +165,29 @@
         this.catalogs = null;
 
         /**
-         * should be used to log information on the server to different log locations
+         * should be used to log client action information on the server
          * @param {Object} headers - the headers to be logged, should include action
-         * @param {String} location - the path for logging (terminal_error || button_action)
          **/
-        this.logHeaders = function (actionHeader, location) {
+        this.logClientAction = function (contextHeaderParams) {
             var defer = module._q.defer();
 
+            if (!contextHeaderParams || (contextHeaderParams === Object(contextHeaderParams) && !Array.isArray(contextHeaderParams))) {
+                var error = new module.InvalidInputError("Context header params were not passed");
+                // Errors for client action logging should not force a terminal error
+                defer.resolve(error);
```

Comment:
You need to return the promise here too. If not, it will run the rest of the function as well. Also, it's better if you reject the promise instead of resolve. You could write this in one line like the following:

```javascript
return defer.reject(error), defer.promise;
```